setup.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) 2016-2018, The University of Texas at Austin
+# & University of California, Merced.
+#
+# All Rights reserved.
+# See file COPYRIGHT for details.
+#
+# This file is part of the hIPPYlib library. For more information and source code
+# availability see https://hippylib.github.io.
+#
+# hIPPYlib is free software; you can redistribute it and/or modify it under the
+# terms of the GNU General Public License (as published by the Free
+# Software Foundation) version 2.0 dated June 1991.
+
+
+from setuptools import setup, find_packages
+from os import path
+from io import open
+
+
+try:
+    from dolfin import __version__ as dolfin_version
+except ImportError:
+    raise
+else:
+    if (int(dolfin_version[0:4]) < 2016):
+        raise Exception(
+            "hippylib requires FEniCS installation not older than 2016.1.0")
+
+here = path.abspath(path.dirname(__file__))
+with open(path.join(here, 'README.md'), encoding='utf-8') as f:
+    long_description = f.read()
+
+REQUIREMENTS = [
+    "mpi4py",
+    "numpy",
+    "matplotlib",
+    "scipy",
+    "sympy==1.1.1",
+    "jupyter"
+]
+
+setup(
+    name='hippylib',
+    version='1.7.0.dev',
```

Comment:
It should be `'2.2.0.dev'`.
At this time the `hIPPYlib` version appears in three different locations:
- this `setup.py`
- `hippylib/__init__.py`
- `doc/config.py`

I'm wondering if there is an easy way to keep those consistent.

Reply:
Just a thought- one may want to make Jupyter an optional dependency and provide python file variants of the jupyter notebooks?

Some may not want jupyter (it can be annoying to install, etc) and it’s not strictly needed for hippylib. Only for examples.

In the same line of reasoning, matplotlib can also be optional. Users who already are farmiliar with hippylib who simply want to use its functionality don’t “need” matplotlib nor jupyter.

Reply:
@uvilla Is there a way to get hippylib github organization email? I would put it there so all the official maintainers get access to it?

Reply:
@eldarkh ,

I don't think github grants e-mail addresses.
We may ask RT at ICES to create a mail redirect for `hippylib-admin@ices.utexas.edu` that redirect to our e-mail addresses.

Reply:
Ok, I will use your email for now, and we change it later if need be